dragonfly/story.py

```diff
@@ -0,0 +1,262 @@
+            room_polygons = tuple(room_polygons)
+        assert len(room_polygons) > 0, 'Story must have at least one RoomPolygon.'
+        for room in room_polygons:
+            assert isinstance(room, RoomPolygon), \
+                'Expected dragonfly RoomPolygon. Got {}'.format(type(room))
+            room._parent = self
+        self._room_polygons = room_polygons
+
+        # check the story geometry that it's within the same plane
+        if tolerance is not None:
+            z_vals = tuple(rm.floor_height for rm in self._room_polygons)
+            assert max(z_vals) - min(z_vals) <= tolerance, 'Not all of Story ' \
+                '"{}" room polygons lie within the same horizontal plane.'.format(name)
+            self._tolerance = tolerance
+        else:
+            self._tolerance = 0
+
+        # process the input properties
+        self.floor_to_floor_height = floor_to_floor_height
+        self.is_ground_floor = is_ground_floor
+        self.is_top_floor = is_top_floor
+
+        self._parent = None  # _parent will be set when Story is added to a Building
+        self._properties = StoryProperties(self)  # properties for extensions
+
+    @property
+    def room_polygons(self):
+        """A tuple of RoomPolygon objects that form an entire story of a building."""
+        return self._room_polygons
+
+    @property
+    def floor_to_floor_height(self):
+        """Get or set a number for the distance from this floor plate to the next one."""
+        return self._floor_to_floor_height
+
+    @floor_to_floor_height.setter
+    def floor_to_floor_height(self, value):
+        if value is None:
+            value = max([room.floor_to_ceiling_height for room in self._room_polygons])
+        self._floor_to_floor_height = float_positive(value, 'floor-to-floor height')
+
+    @property
+    def is_ground_floor(self):
+        """Get or set a boolean to note whether this Story is a ground floor."""
+        return self._is_ground_floor
+
+    @is_ground_floor.setter
+    def is_ground_floor(self, value):
+        self._is_ground_floor = bool(value)
+
+    @property
+    def is_top_floor(self):
+        """Get or set a boolean to note whether this Story is a top floor."""
+        return self._is_top_floor
+
+    @is_top_floor.setter
+    def is_top_floor(self, value):
+        self._is_top_floor = bool(value)
+
+    @property
+    def parent(self):
+        """Parent Building if assigned. None if not assigned."""
+        return self._parent
+
+    @property
+    def has_parent(self):
+        """Boolean noting whether this Story has a parent Building."""
+        return self._parent is not None
+
+    @property
+    def geometry(self):
+        """A ladybug_geometry Polyface3D object representing the floor plate."""
+        room_floors = [room.geometry for room in self.room_polygons]
+        return Polyface3D.from_faces(room_floors, self._tolerance)
+
+    @property
+    def outline_curves(self):
```

Comment:
I suggest `polyline` instead of `outline_curves`.

Reply:
It definitely needs to be plural since you always get a list of curves in the case that the story has a holes in the floor plate. However, when someone requests `polylines` it sounds like they might get a polyline for each `Room2D`. Not the outline of the entire floor, which is what this property is returning. You're right that `curves` is not the right term (I'm definitely stuck in Rhino SDK vocabulary here). What about `boundary_polylines`?

Reply:
I think that I will change it to `outline_segments` since I think it is actually useful to not have them joined into a polyline. So this will return a list of `LineSegment3D` objects outlining the shape, which can be extruded to represent the context shading of the building. This way, I can implement this feature without needing to implement polylines in ladybug-geometry.

Reply:
+1 for `outline_segments`

Reply:
Great! I also changed it to be a method instead of a property particularly since I realized that a `tolerance` input was important for it to work correctly.